packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp

```diff
@@ -260,6 +262,25 @@ void GSIterate(RCP<Teuchos::ParameterList> smootherParams,
       OneregX[k] += ldelta[k];
     }
     zeroInitGuess = false;
+
+    if( sgs ){
+      for (size_t k = numRows; k--;) {
```

Comment:
Shouldn't this be
```suggestion
      for (size_t k = numRows - 1; k >= 0; k--) {
```
to start looping at the last row and then move backward to the first row?

Reply:
I'm not sure why, but `for (size_t k = numRows - 1; k >= 0; k--)` does not work. It is some sort of c++ thing that I don't fully understand.
I did check that `for (size_t k = numRows; k--;)` correctly iterates from last row to first row (starts at `numRows-1` and ends at `0`).

Reply:
I was just curious. If it works, then I'm fine with it.

Reply:
@pohm01 @mayrmt size_t is unsigned, so if it's 0 and you decrement it, it wraps around to 2^64 - 1. So k >= 0 is true no matter what k is. You could use signed counter, or loop k from 0 to numRows and then use ``numRows - k - 1`` in the loop.